### PR TITLE
Cart & wishlist functionality

### PR DESCRIPTION
**Pull request by parulgupta2001**
Add to cart & Add to wishlist functionality, footer

### CODE REVIEW

File: my-app/src/frontend/components/navbar/Navbar.js [hunk 1]
```diff
@@ -4,12 +4,14 @@ import { AiFillHeart } from "react-icons/ai";
 import "./navbar.css";
 import "../../../App.css";
 import { Link, useNavigate } from "react-router-dom";
-import { useAuth } from "../../contexts/auth-context";
+import { useAuth, useCart, useWishlist } from "../../contexts/index";
 
 function Navbar() {
   const { authState, authDispatch } = useAuth();
```
Comment (poojavishnoi) [inline my-app/src/frontend/components/navbar/Navbar.js:10]: you can deconstrute token here itself, you just have to pass token directly from auth-context.

File: my-app/src/frontend/components/products/Products.js [hunk 2]
```diff
@@ -1,25 +1,141 @@
...
-        <div className="img-card-container">
-          <AiOutlineHeart />
-          <img className="img-container" src={img} />
-          <div> {name} </div>
-          <div>Rs. {price}</div>
-          <div> ★{rating} </div>
-          <div className="btn-container">
-            <button className="add-to-cart-btn btn">ADD TO CART</button>
+      {categoryData.map(
+        ({ img, _id, name, price, rating, stock, delivery, categoryName }) => (
```
Comment (poojavishnoi) [inline my-app/src/frontend/components/products/Products.js:64]: Nice that you have deconstruted here only.

File: my-app/src/frontend/contexts/cart-context.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+const CartContext = createContext();
+
+const CartProvider = ({ children }) => {
+  const [cartItems, setCartItems] = useState([]);
+
+  useEffect(() => {
+    axios.get("/api/user/cart").then((response) => {
+      setCartItems(response.data.cart);
+      console.log(response.data.cart);
```
Comment (poojavishnoi) [inline my-app/src/frontend/contexts/cart-context.js]: you can remove console.log

File: my-app/src/frontend/pages/home/Home.js [hunk 4]
```diff
@@ -1,12 +1,14 @@
 import { Filter } from "../../components/filter/Filter";
 import { Products } from "../../components/products/Products";
+import { Footer } from "../../components/footer/Footer";
```
Comment (poojavishnoi) [inline my-app/src/frontend/pages/home/Home.js]: You can have one index.js in component and there export all the components, then here you can import all components in one line.

**Review by poojavishnoi — APPROVED** [review]
Left few suggestions.Code is clean.Good work

File: my-app/src/frontend/pages/wishlist/wishlist.css [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+} 
+
+
+
+.wishlist-img-container {
+    height:7rem;
+}
+
+.wishlist-container-upper{
+   transform: translate(80px,-600px);
```
Comment (rutu14) [inline my-app/src/frontend/pages/wishlist/wishlist.css:23]: Suggestion: Can change to rems

**Review by rutu14 — APPROVED** [review]
It is good to merge

**Pull request merged**